packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,323 @@
+        const ext = "."  + (commandParameters.arguments.extension ?? "txt");
+        if (lfFile.fileType == 'uss'){
+            // Hash in a repeatable way if uss fileName (incase there are special characters in name)
+            const crypto = require("crypto");
+            const hash = crypto.createHash('sha256').update(lfFile.fileName).digest('hex');
+            return path.join(tmpdir(), hash + ext);
+        }
+        return path.join(tmpdir(), lfFile.fileName + ext);
+    }
+
+    /**
+     * Check for temp path's existence (check if previously `stashed` edits exist)
+     * @param {string} tempPath - unique file path for local file (stash/temp file)
+     * @returns {Promise<boolean>} - promise that resolves to true if stash exists or false if doesn't
+     * @memberof EditUtilities
+     */
+    public static async checkForStash(tempPath: string): Promise<boolean>{
+        try {
+            return existsSync(tempPath);
+        } catch(err) {
+            throw new ImperativeError({
+                msg: `Failure when checking for stash. Command terminated.`,
+                causeErrors: err
+            });
+        }
+    }
+
+    /**
+     * Collection of prompts to be used at different points in editing process
+     * @param {Prompt} prompt - selected prompt from {@link Prompt} (enum object)
+     * @param {string} tempPath - unique file path for local file (stash/temp file)
+     * @returns {Promise<boolean>} - promise that resolves depending on prompt case and user input
+     * @memberof EditUtilities
+     */
+    public static async promptUser(prompt: Prompt, tempPath?: string): Promise<boolean>{
+        let input;
+        switch (prompt){
+            case Prompt.useStash:
+                input = await CliUtils.readPrompt(TextUtils.chalk.green(`Keep and continue editing found stash? Y/n`));
+                if (input === null) {
+                    throw new ImperativeError({
+                        msg: `No input provided. Command terminated.`
+                    });
+                }
+                return input.toLowerCase() === 'y';
+            case Prompt.doneEditing:
+                do{
+                    input = await CliUtils.readPrompt(TextUtils.chalk.green(`Enter "done" in terminal once finished `+
+                    `editing and saving temporary file: ${tempPath}`), {secToWait: 3600});
+                }while(input.toLowerCase() !== 'done');{
+                    if (input === null) {
+                        throw new ImperativeError({
+                            msg: TextUtils.chalk.red(`No input provided. Command terminated. Stashed file will persist: ${tempPath}`)
+                        });
+                    }
+                    return true;
+                }
+            case Prompt.continueToUpload:
+                input = await CliUtils.readPrompt(TextUtils.chalk.green(`Continue uploading edits despite changes on remote? Y/n`));
+                if (input === null) {
```

Comment:
Are you testing for a trinary decision with the "input == null" tests in this function? That is to say:  yes, no, or abort?

If you really only want a yes/no answer, you could loop on the prompts until you get a yes/not-yes response, and never throw an exception.

Reply:
If null that means prompt was timed out, which is the cue to cancel the command rather than leaving it running in someones terminal forever. but maybe we never want to time out?